src/components/Modal/index.tsx

```diff
@@ -1,75 +1,87 @@
 import Center from '@components/@layout/Center';
-import Flexbox from '@components/@layout/Flexbox';
-import Typography from '@components/Typography';
+import Portal from '@components-common/Portal';
+import { PORTAL_MODAL_ROOT_ID } from '@constants/portal';
 import styled from '@emotion/styled';
+import { ChildrenProps } from '@util-types/ChildrenProps';
 import { DefaultPropsWithChildren } from '@utils/types/DefaultPropsWithChildren';
-import { MouseEventHandler } from 'react';
+import { MouseEventHandler, createContext } from 'react';
+import { MdArrowBackIosNew } from 'react-icons/md';
+import useSafeContext from 'src/hooks/useSafeContext';
 
-type Action = { key: string; handler: MouseEventHandler };
 interface ModalProps extends DefaultPropsWithChildren<HTMLDivElement> {
-  title?: string;
   isOpen: boolean;
   onClose: MouseEventHandler;
-  actions?: Action[];
 }
 
-const Modal = ({ title, children, isOpen, onClose, actions }: ModalProps) => {
+interface HeaderProps {
+  title: string;
+}
+interface ModalContextInterface {
+  onClose: MouseEventHandler;
+}
+
+const ModalContext = createContext<ModalContextInterface | null>(null);
+
+const Modal = ({ children, isOpen, onClose }: ModalProps) => {
+  const contextValues = {
+    onClose,
+  };
   return (
-    <ModalWrapper {...{ isOpen }}>
-      <BackGround onClick={onClose}></BackGround>
-      <Center>
-        <ModalBox>
-          {title && <Typography variant="subtitle1">{title}</Typography>}
-          <Content>{children}</Content>
-          {actions && (
-            <Footer>
-              <Flexbox justifyContent="space-evenly">
-                {actions.map(({ key, handler }) => (
-                  <Button key={key} onClick={handler}>
-                    {key}
-                  </Button>
-                ))}
-              </Flexbox>
-            </Footer>
-          )}
-        </ModalBox>
-      </Center>
-    </ModalWrapper>
+    <Portal id={PORTAL_MODAL_ROOT_ID}>
+      <ModalContext.Provider value={contextValues}>
+        <ModalWrapper {...{ isOpen }}>
+          <BackGround onClick={onClose} />
+          <Center>
+            <ModalBox role="dialog" aria-modal="true">
+              {children}
+            </ModalBox>
+          </Center>
+        </ModalWrapper>
+      </ModalContext.Provider>
+    </Portal>
   );
 };
 
+const Header = ({ title }: HeaderProps) => {
+  const { onClose } = useSafeContext(ModalContext);
+  return (
+    <HeaderWrapper>
+      <Button onClick={onClose}>
+        <Center>
+          <MdArrowBackIosNew />
+        </Center>
+      </Button>
+      <Title>{title}</Title>
+    </HeaderWrapper>
```

Comment:
```suggestion
<Flexbox
      justifyContent='left'
      css={css`
        height: 50px;
        padding: 20px;
        border-bottom: 1px solid ${themeColor.gray100};
      `}
    >
      <Button onClick={onClose}>
        <Center>
          <MdArrowBackIosNew />
        </Center>
      </Button>
      <Typography variant="body">{title}</Typography>
    </Flexbox>
```

기존에 있는 컴포넌트들을 활용해보았어요.

align-items랑 line-height 는 각가 삭제하거나 변경해도 변화가 눈에 안보여서 없애보았는데,

한 번 검토 부탁드려요!

Reply:
옹 잘 동작해요!! 감사해요 반영할게요!!